services/api/lib/bulk.js

```diff
@@ -152,21 +152,39 @@ exports.register = function(server, options, done) {
   // 0th index and the value on that object keyed with 'id'
   var reachRegex = /^\$(\d+)\.(.*)$/;
 
+  var pipelineKeys = [
+    'id',
+    'projectId',
+    'pageId'
+  ];
+
   // check if a key on the action object should be resolved to a value
   // returned by a previous action in the transaction
-  function generateForEachCallback(processResult, action, actionIndex, txResults) {
-    return function everyCallback(key) {
-      // if the key's value isn't a reach string, return valid
-      if ( !reachRegex.test(action.data[key]) ) {
-        return processResult;
+  function reachForData(action, actionIndex, txResults) {
```

Comment:
this is still a very large function- do we want to try to simplify it into smaller util functions? i think the iteration (lines 166-171) could be one, also the replace function in lines 182-187.

this way this function is really just 2 functions and some checks. much easier to reason about i think (function names help explain what is going on)